### ISSUE TITLE
Chore: Deploy Demo App to Fly.io

### ISSUE THREAD

**Issue by nelsonic** [dwyl/elixir-auth-google-demo#15]
Heroku has kindly deleted our old demo app - that was consuming 0.0001 hours per month and basically cost them _nothing_ ...
https://elixir-auth-google-demo.herokuapp.com/
<img width="770" alt="image" src="https://user-images.githubusercontent.com/194400/217832955-39d62a13-6b7c-4465-bd3b-09a8c88983f8.png">

So ...

# Todo

+ [x] Deploy this app to Fly.io

The _reason_ I'm opening this now is to help debug: https://github.com/dwyl/elixir-auth-google/issues/83 💭 
Which in turn will help ensure that our `auth` is working correctly. 👌 

Related to: https://github.com/dwyl/learn-heroku/issues/52

**Comment by nelsonic** [comment]
Failed:

```sh
Preparing system for Elixir builds
Installing application dependencies
Running Docker release generator
Error failed running /opt/homebrew/bin/mix phx.gen.release --docker: exit status 1
```

**Comment by nelsonic** [comment]
Error while trying to deploy to `Fly.io`:
```sh
--> Pushing image done
image: registry.fly.io/elxir-google-auth-demo:deployment-01GRVCAZJD0573G1F8NGQ0Z7WH
image size: 122 MB
==> Creating release
--> release v2 created

--> You can detach the terminal anytime without stopping the deployment
==> Monitoring deployment
Logs: https://fly.io/apps/elxir-google-auth-demo/monitoring

 1 desired, 1 placed, 0 healthy, 1 unhealthy [restarts: 2] [health checks: 1 total]
Failed Instances

Failure #1

Instance
ID      	PROCESS	VERSION	REGION	DESIRED	STATUS 	HEALTH CHECKS	RESTARTS	CREATED 
eedf95a6	app    	0      	lhr   	run    	pending	1 total      	2       	25s ago	

Recent Events
TIMESTAMP           	TYPE      	MESSAGE                         
2023-02-09T15:22:46Z	Received  	Task received by client        	
2023-02-09T15:22:46Z	Task Setup	Building Task Directory        	
2023-02-09T15:22:55Z	Started   	Task started by client         	
2023-02-09T15:22:59Z	Terminated	Exit Code: 1                   	
2023-02-09T15:22:59Z	Restarting	Task restarting in 1.186411976s	
2023-02-09T15:23:05Z	Started   	Task started by client         	
2023-02-09T15:23:09Z	Terminated	Exit Code: 1                   	
2023-02-09T15:23:09Z	Restarting	Task restarting in 1.005551008s	

2023-02-09T15:23:07Z   [info]    (elixir 1.14.3) lib/code.ex:425: Code.validated_eval_string/3
2023-02-09T15:23:07Z   [info]5,110,103,46,10,70,111,114,32,101,120,97,109,112,108,101,58,32,101,99,116,111,58,47,47,85,83,69,82,58,80,65,83,83,64,72,79,83,84,47,68,65,84,65,66,65,83,69,10>>},[{elixir_eval,'__FILE__',1,[{file,"/app/releases/1.6.5/runtime.exs"},{line,26}]},{elixir,'-eval_external_handler/1-fun-2-',4,[{file,"src/elixir.erl"},{line,309},{error_info,#{module=>'Elixir.Exception'}}]},{erl_eval,do_apply,7,[{file,"erl_eval.erl"},{line,748}]},{erl_eval,expr,6,[{file,"erl_eval.erl"},{line,492}]},{erl_eval,exprs,6,[{file,"erl_eval.erl"},{line,136}]},{elixir,eval_forms,4,[{file,"src/elixir.erl"},{line,294}]},{'Eli
2023-02-09T15:23:07Z   [info]5,110,103,46,10,70,111,114,32,101,120,97,109,112,108,101,58,32,101,99,116,111,58,47,47,85,83,69,82,58,80,65,83,83,64,72,79,83,84,47,68,65,84,65,66,65,83,69,10>>},[{elixir_eval,'__FILE__',1,[{file,"/app/releases/1.6.5/runtime.exs"},{line,26}]},{elixir,'-eval_external_handler/1-fun-2-',4,[{file,"src/elixir.erl"},{line,309},{error_info,#{module=>'Elixir.Exception'}}]},{erl_eval,do_apply,7,[{file,"erl_eval.erl"},{line,748}]},{erl_eval,expr,6,[{file,"erl_eval.erl"},{line,492}]},{erl_eval,exprs,6,[{file,"erl_eval.erl"},{line,136}]},{elixir,eval_forms,4,[{file,"src/elixir.erl"},{line,294}]},{'Eli
ir.Module.ParallelChecker',verify,1,[{file,"lib/module/parallel_checker.ex"},{line,110}]},{'Elixir.Code',validated_eval_string,3,[{file,"lib/code.ex"},{line,425}]}]}}
2023-02-09T15:23:07Z   [info]Crash dump is being written to: erl_crash.dump...done
2023-02-09T15:23:08Z   [info]Starting clean up.
2023-02-09T15:23:13Z   [info]Starting instance
2023-02-09T15:23:13Z   [info]Configuring virtual machine
2023-02-09T15:23:13Z   [info]Pulling container image
2023-02-09T15:23:13Z   [info]Unpacking image
2023-02-09T15:23:13Z   [info]Preparing kernel init
2023-02-09T15:23:14Z   [info]Configuring firecracker
2023-02-09T15:23:14Z   [info]Starting virtual machine
2023-02-09T15:23:14Z   [info]Starting init (commit: c19b424)...
2023-02-09T15:23:14Z   [info]Preparing to run: `/app/bin/server` as nobody
2023-02-09T15:23:14Z   [info]2023/02/09 15:23:14 listening on [fdaa:1:3fd8:a7b:bbfa:eedf:95a6:2]:22 (DNS: [fdaa::3]:53)
2023-02-09T15:23:17Z   [info]ERROR! Config provider Config.Reader failed with:
2023-02-09T15:23:17Z   [info]** (RuntimeError) environment variable DATABASE_URL is missing.
2023-02-09T15:23:17Z   [info]For example: ecto://USER:PASS@HOST/DATABASE
2023-02-09T15:23:17Z   [info]    /app/releases/1.6.5/runtime.exs:26: (file)
2023-02-09T15:23:17Z   [info]    (elixir 1.14.3) src/elixir.erl:309: anonymous fn/4 in :elixir.eval_external_handler/1
2023-02-09T15:23:17Z   [info]    (stdlib 4.2) erl_eval.erl:748: :erl_eval.do_apply/7
2023-02-09T15:23:17Z   [info]    (stdlib 4.2) erl_eval.erl:492: :erl_eval.expr/6
2023-02-09T15:23:17Z   [info]    (stdlib 4.2) erl_eval.erl:136: :erl_eval.exprs/6
2023-02-09T15:23:17Z   [info]    (elixir 1.14.3) src/elixir.erl:294: :elixir.eval_forms/4
2023-02-09T15:23:17Z   [info]    (elixir 1.14.3) lib/module/parallel_checker.ex:110: Module.ParallelChecker.verify/1
2023-02-09T15:23:17Z   [info]    (elixir 1.14.3) lib/code.ex:425: Code.validated_eval_string/3
2023-02-09T15:23:17Z   [info]init terminating in do_boboot",{#{'__exception__'=>true,'__struct__'=>'Elixir.RuntimeError',message=><<101,110,118,105,114,111,110,109,101,110,116,32,118,97,114,105,97,98,108,101,32,68,65,84,65,66,65,83,69,95,85,82,76,32,105,115,32,109,105,115,115,105,110,103,46,10,70,111,114,32,101,120,97,109,112,108,101,58,32,101,99,116,111,58,47,47,85,83,69,82,58,80,65,83,83,64,72,79,83,84,47,68,65,84,65,66,65,83,69,10>>},[{elixir_eval,'__FILE__',1,[{file,"/app/releases/1.6.5/runtime.exs"},{line,26}]},{elixir,'-eval_external_handler/1-fun-2-',4,[{file,"src/elixir.erl"},{line,309},{error_info,#{module=>'Elixir.Exception'}}]},{erl_eval,do_apply,7,[{file,"erl_eval.erl"},{line,748}]},{erl_eval,expr,6,[{file,"erl_eval.erl"},{line,492}]},{erl_eval,exprs,6,[{file,"erl_eval.erl"},{line,136}]},{elixir,eval_forms,4,[{file,"src/elixir.erl"},{line,294}]},{'Elixir.Module.ParallelChecker',verify,1,[{file,"lib/module/parallel_checker.ex"},{line,110}]},{'Elixir.Code',validated_eval_string,3,[{file,"lib/code.ex"},{line,425}]}]}}
2023-02-09T15:23:17Z   [info]init terminating in do_boot ({,[{elixir_eval,__FILE__,1,[{_},{_}]},{elixir,-eval_external_handler/1-fun-2-,4,[{_},{_},{_}]},{erl_eval,do_apply,7,[{_},{_}]},{erl_eval,expr,6,[{_},{_}]},{erl_eval,exprs,6,[{_},{_}]},{elixir,eval_forms,4,[{_},{_}]},{Elixir.Module.ParallelChecker,verify,1,[{_},{_}]},{Elixir.Code,validated_eval_string,3,[{_},{_}]}]})
2023-02-09T15:23:17Z   [info]Crash dump is being written to: erl_crash.dump...done
2023-02-09T15:23:17Z   [info]Starting clean up.
--> v0 failed - Failed due to unhealthy allocations - no stable job version to auto revert to and deploying as v1 

--> Troubleshooting guide at https://fly.io/docs/getting-started/troubleshooting/
Error abort
```

Also, on localhost getting the following error: 🤷‍♂️ 
```
spawn: Could not cd to /elixir-auth-google-demo/assets
```

**Comment by nelsonic** [comment]
```sh
fly secret set SECRET_KEY_BASE=wLJ3lRg5sbJSD3G4IOoqDBnG0+iSyXWyu+bChlHMsJwoC3i4c0sXUwKretc.
```

**Comment by nelsonic** [comment]
Getting the following error when trying to write a test:
```sh
%{error: "invalid_grant", error_description: "Malformed auth code."}
```

**Comment by nelsonic** [comment]
https://elixir-google-auth-demo.fly.dev/
<img width="823" alt="image" src="https://user-images.githubusercontent.com/194400/217882666-fc68e48e-8204-433b-95f8-e0fb9d119c8b.png">

502 ... 🤷‍♂️ 
<img width="914" alt="image" src="https://user-images.githubusercontent.com/194400/217885189-b7276137-af1c-4be7-98fa-a8faa919f0be.png">

**Comment by nelsonic** [comment]
Working:

<img width="836" alt="image" src="https://user-images.githubusercontent.com/194400/217935199-2aa46e54-6977-4333-a3ac-22feab777004.png">

**Comment by nelsonic** [comment]
Deployed and documented. PR: #16 ✅

**Comment by SimonLab** [comment]
The github action to deploy the application is not properly formatted:
![image](https://user-images.githubusercontent.com/6057298/218440976-054fbba7-e85d-46dd-b7dc-5050b562d62d.png)

The `deploy` action needs to be indented to be under the `jobs` section

**Comment by nelsonic** [comment]
Should be fixed now. 👍

**Comment by nelsonic** [comment]
Fixed by #19 ✅
